### PR TITLE
Add requirements to meet WordPress.org block directory guidelines

### LINKED ISSUE #62: Prepare for Block Directory
**Is your enhancement related to a problem? Please describe.**
[Gutenberg 8.4](https://make.wordpress.org/core/2020/06/24/whats-new-in-gutenberg-24-june/) officially enables the Block Directory which means this will officially ship in [WordPress 5.5](https://make.wordpress.org/core/5-5/) on/around August 11th.  In order to support installs on Gutenberg 8.4+ and eventual WordPress 5.5+, we'll want to ensure Block for Apple Maps adheres to the Block Directory standards to ensure the block appears when searched for.  Note that in my current testing searching for "apple maps", "apple", and "maps" does not return results.

**Describe the solution you'd like**

- [ ] ensure plugin adheres to [Block Directory submission guidelines](https://github.com/WordPress/wporg-plugin-guidelines/blob/block-guidelines/blocks.md) ([see also](https://github.com/WordPress/gutenberg/pull/23545))
- [ ] pass [Block Plugin Validator](https://wordpress.org/plugins/developers/block-plugin-validator/) (current issues listed below):
   - [x] No block tag found in readme.txt. [More about this.](https://wordpress.org/plugins/developers/block-plugin-validator/help#check_block_tag)
   - [x] No block.json files were found. [More about this.](https://wordpress.org/plugins/developers/block-plugin-validator/help#check_for_block_json)
   - [ ] No scripts found for block tenup/maps-block-apple. [More about this.](https://wordpress.org/plugins/developers/block-plugin-validator/help#check_for_block_scripts)
   - [ ] Found 2 PHP asset files. [More about this.](https://wordpress.org/plugins/developers/block-plugin-validator/help#check_asset_php_file)
   - [ ] Found 18 KB of PHP code. This might not be a block plugin. [More about this.](https://wordpress.org/plugins/developers/block-plugin-validator/help#check_php_size)

**Designs**

n/a

**Describe alternatives you've considered**

n/a

**Additional context**

[Open issues for Gutenberg specific to the Block Directory.](https://github.com/wordpress/gutenberg/issues?q=is%3Aopen+is%3Aissue+label%3A%22%5BFeature%5D+Block+Directory%22)
[Block Validator Help.](https://wordpress.org/plugins/developers/block-plugin-validator/help/)

### PR DESCRIPTION
**Pull request by jeffpaul**
### Description of the Change

#62 outlines what's needed fairly well, this PR adds the `block` tag to `readme.txt` and the `block.json` file.  I'm leaving this as a draft PR as the `block.json` still needs the scripts/styles added to the end of it as well as validating all other fields that I've attempted to complete.  Note that there's duplication of the `attributes` and `icon` in the `block.json` with other places in the plugin codebase as I copied those values into the `block.json` file, I'm not certain if we can or should remove other references as the block submission guidelines and other details are still a bit in flux.

### Alternate Designs

n/a

### Benefits

Ensures that this plugin appears in the Block Directory search within Gutenberg.

### Possible Drawbacks

Given that we've got PHP code for the Settings page, we may need to get manual approval from the Plugin Review Team as there's supposed to be minimal-to-no PHP code in "block plugins".

### Verification Process

TBD as this is still WIP, haven't verified anything fully yet.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Closes #62.

### Changelog Entry

```
### Added
- WordPress.org Block Directory integration
```

### CODE REVIEW

**Comment by fabiankaegy** [comment]
@jeffpaul My main concern is the custom REST endpoint. And we would need to remove the settings page. Because there can be no UI outside the editor

**Comment by jeffpaul** [comment]
@fabiankaegy yes, the "no PHP" stance is a current issue for us though I'm discussing with folks over in core about this in cases for blocks that need some setup/config steps that ideally happens in a Settings page.  I know that the Plugin Review Team can add a block plugin to the Block Directory themselves, so I'm hoping our case at worst is an exception to the rule until the Gutenberg team is able to provide a better way to handle block setup/config akin to a Settings page.  For now let's keep our settings as-is and only change if the Plugin Review Team and Gutenberg team say we _must_ change.

**Comment by jeffpaul** [comment]
@fabiankaegy @ryanwelcher I could use your help in confirming the styles/scripts to reference in the block.json file, noting details on these potential four entries starts here: https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#editor-script

File: block.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+	"name": "tenup/maps-block-apple",
+	"title": "Apple Maps",
+	"category": "location",
```
Comment (jeffpaul) [inline block.json]: This category appears to be a custom one, maybe we should update to `embeds` to use one of the default block categories? cc: @fabiankaegy

**Comment by JohnRDOrazio** [comment]
I just ran my plugin through the new [Block Checker](https://wordpress.org/plugins/developers/block-plugin-validator/) and it's giving a `Fatal Error` with this reason: `Found PHP call header(). This is likely to prevent your plugin from working as expected.` It's referring to this code at the top of the plugin:
`if (!defined('ABSPATH')) {
  header('Status: 403 Forbidden');
  header('HTTP/1.1 403 Forbidden');
  exit();
}`

It was in my understanding that this is always a good thing to put at the top of a plugin or theme in order to avoid direct access. Why would this be considered a `Fatal Error` ?

File: block.json (hunk 1 above)
Comment (jeffpaul) [reply block.json]: Looks like `embed` is the correct spelling based on https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-registration.md#category

**Comment by jeffpaul** [comment]
Sharing a screenshot from @fabiankaegy of request changes, I'll update for these now. Screen Shot 2020-08-11 at 18 33 00

**Comment by jeffpaul** [comment]
Per https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-metadata.md it looks like we'll need to keep the `file:` prepended to the scripts/styles at the end of the `block.json` file.

**Pull request merged**